src/smt.rs

```diff
@@ -149,43 +149,15 @@ pub struct SparseMerkleTree {
     hasher: Hasher,
 }
 
-fn key_hash(key: &[u8]) -> Vec<u8> {
-    let prefix = &key[..PREFIX_SIZE];
-    let body = &key[PREFIX_SIZE..];
-    let mut hasher = Sha256::new();
-    hasher.update(body);
-    let result = hasher.finalize();
-    [prefix, result.as_slice()].concat()
-}
-
-fn value_hash(value: &[u8]) -> Vec<u8> {
-    let mut hasher = Sha256::new();
-    hasher.update(value);
-    let result = hasher.finalize();
-    result.to_vec()
-}
-
-fn leaf_hash(pair: &KVPair) -> Vec<u8> {
-    let mut hasher = Sha256::new();
-    hasher.update(PREFIX_LEAF_HASH);
-    hasher.update(pair.key());
-    hasher.update(pair.value());
-    let result = hasher.finalize();
-    result.to_vec()
-}
-
-fn branch_hash(node_hash: &[u8]) -> Vec<u8> {
-    let mut hasher = Sha256::new();
-    hasher.update(PREFIX_BRANCH_HASH);
-    hasher.update(node_hash);
-    let result = hasher.finalize();
-    result.to_vec()
-}
-
-fn empty_hash() -> Vec<u8> {
-    let hasher = Sha256::new();
-    let result = hasher.finalize();
-    result.to_vec()
+impl Hash256 for KVPair {
```

Comment:
this Hash256 implementation is quite specific to the SMT implementation while KVPair is quite generic type?

Reply:
Yes, We need a specific implementation of the `Hash256` for `KVPair` to use it in `SMT`.

Reply:
I meant `KVPair` is quite generic type while it has specific SMT implementation seemed bit strange but it's under `smt` module so very minor point 👍